SCClassLibrary/Common/Core/Boolean.sc

```diff
@@ -18,9 +18,13 @@ Boolean {
 	keywordWarnings {
 		// turn on/off warnings if a keyword argument is not found
 		_KeywordError
+		^this.primitiveFailed
+	}
+	trace {
+		// this is only available in a special debugging version of the app
```

Comment:
^ this is the problem though, that everyone has a different idea of what's OK to compromise on, and what isn't. For instance, in this PR, I was more than happy to compromise and let this topic go. But in the larger scheme, I think it does make the codebase messy and significantly more difficult to read, and I would push for it to be changed.

While I was reviewing this, I spent at least a couple seconds reading comments that I wouldn't have read otherwise because they looked like they signalled something unusual about the function body, rather than harmlessly describing the function itself. Like almost everyone else (see [this thread](https://softwareengineering.stackexchange.com/questions/126601/comment-before-or-after-the-relevant-code) for instance), and like most code documentation schemes (Javadoc, Rustdoc, Doxygen, etc.), I am conditioned to expect comments to apply to the code below, or the the statement to the left, and not the code above. So, it was difficult to read because, even though nothing was amiss, I had to read the comment, understand that it was descriptive, and then continue on to the next function body, in the same time that I could have scanned multiple visually similar function blocks. It was distracting, and not just because I wanted to correct it :)

Unlike spacing around parens, putting a comment in the wrong place breaks semantic flow of the document and also introduces the potential for confusion. That's why I'd consider it a step higher in importance.

I'm aware that this sounds myopic out of context, but I just want to lay out very cleanly and clearly why I made my initial comment.